chromium_src/chrome/browser/ui/views/tabs/alert_indicator_button.cc

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "chrome/browser/ui/views/tabs/alert_indicator_button.h"
+
+#include "brave/components/constants/pref_names.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/views/tabs/tab.h"
+#include "chrome/browser/ui/views/tabs/tab_slot_controller.h"
+#include "components/prefs/pref_service.h"
+
+#define AlertIndicatorButton AlertIndicatorButtonBase
+
+#include "src/chrome/browser/ui/views/tabs/alert_indicator_button.cc"
+
+#undef AlertIndicatorButton
+
+void AlertIndicatorButton::UpdateEnabledForMuteToggle() {
+  const bool was_enabled = GetEnabled();
+  auto* profile = GetTab()->controller()->GetBrowser()->profile();
+  const bool pref_enabled =
+      profile->GetPrefs()->GetBoolean(kBraveTabsMuteIndicatorClickable);
```

Comment:
I think this pref should be persistent across regular and incognito profile. Can we add this pref to a whitelist in favor of [a thread](https://bravesoftware.slack.com/archives/C7VLGSR55/p1670000442736269)? @simonhong is doing [this](https://github.com/brave/brave-core/pull/16224/files#diff-9c529fa8a0064a7faf1360d953fcfabd2c38787156d9fe57e498e6dfbbad9b7b) and you might be able to use it.

Reply:
++ this could be persistent prefs.

Reply:
Oh cool, didn't know about that! All right, that seems sensible to me. How far off landing do you think that PR is? I might wait for it to land, so I can reuse your util @simonhong

Reply:
I expect my PR could be landed very soon.
but I realized that this PR doesn't need my PR because `kBraveTabsMuteIndicatorClickable` is only changed via normal settings. If private window can change `kBraveTabsMuteIndicatorClickable` directly, then it should be persistent prefs.

Reply:
>  If private window can change kBraveTabsMuteIndicatorClickable directly, then it should be persistent prefs.

Soryy, I think I'm not getting what you mean. Do you mean that we can get proper the value without using GetOrigianlProfile() or persistent pref from incognito windows?

Reply:
As my PR is merged https://github.com/brave/brave-core/pull/16224, we can also make it persistent for prevent future regression. Then, we can touch `kBraveTabsMuteIndicatorClickable` from private profile.

Reply:
> Yes, we can get same value till it's not changed from private profile.

Oh, didn't know that. Sounds like.. copy-on-write strategy for prefs.

Reply:
:+1: